hoot-core/src/main/cpp/hoot/core/io/OsmXmlReader.cpp

```diff
@@ -225,15 +243,24 @@ void OsmXmlReader::_createWay(const QXmlAttributes& attributes)
 
 void OsmXmlReader::_createRelation(const QXmlAttributes& attributes)
 {
-  _relationId = _parseLong(attributes.value("id"));
+  _element.reset();
 
-  // Adding this in causes issues with the tests...worth looking into at some point.
+  long id = _parseLong(attributes.value("id"));
+
+  if (_relationIdMap.contains(id) && _ignoreDuplicates)
+  {
+    LOG_TRACE("Ignoring relation id " << id << " already exists");
+    return;
+  }
+// Adding this in causes issues with the tests...worth looking into at some point.
```

Comment:
I'm going to take a look at this (in another PR) and figure out what is going on.  It should probably be fixed.